LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js

```diff
@@ -338,3 +338,52 @@ window.test_driver_internal.consume_user_activation = async function(context)
         throw new Error("unimplemented");
     return context.internals.consumeTransientActivation();
 }
+
+/**
+ *
+ * @param {Window} context
+ */
+window.test_driver_internal.minimize_window = async function (context=null)
+{
+    if (context === null)
+        context = window;
+    if (context.document?.fullscreenElement)
+        await context.document.exitFullscreen();
+    if (context.document.visibilityState === "hidden")
+        return;
+    return new Promise(resolve => {
+        const rect = internals.visualViewportRect();
+        context.addEventListener("visibilitychange", () => {
+            resolve(rect);
+        }, { once: true });
+        testRunner.setPageVisibility("hidden");
+    });
```

Comment:
Should be `await` unless you meant to return a Promise wrapped inside a Promise.

Or to preserve current behavior, but return undefined, you can just omit `return` from the line

Reply:
Yeah, good suggestion... the method still needs to return the DOMRect, but I can simplify it.

Reply:
`return` is fine but I would keep the style of set_window_rect and minimize_window consistent (await/return and line breaks).